crates/router/src/core/utils.rs

```diff
@@ -1136,6 +1119,34 @@ pub async fn get_profile_id_from_business_details(
     }
 }
 
+#[cfg(all(feature = "v2", feature = "merchant_account_v2"))]
+pub async fn get_profile_id_from_business_details(
```

Comment:
we should not read the profile deep inside this function, where profile is read in core, there this validation should be done.